pkg/api/component_report.go

```diff
@@ -192,21 +195,29 @@ func (c *componentReportGenerator) GenerateVariants() (apitype.ComponentReportTe
 	}, errs
 }
 
-func (c *componentReportGenerator) GenerateReport() (apitype.ComponentReport, []error) {
-	before := time.Now()
-	baseStatus, sampleStatus, errs := c.getTestStatusFromBigQuery()
+func (c *componentReportGenerator) GenerateReport(cc cache.Cache, cacheKey interface{}) (apitype.ComponentReport, []error) {
+	componentReportTestStatus, errs := getDataFromCacheOrGenerate[apitype.ComponentReportTestStatus](cc, c.cacheOption, cacheKey, c.GenerateComponentReportTestStatus, apitype.ComponentReportTestStatus{})
 	if len(errs) > 0 {
 		return apitype.ComponentReport{}, errs
 	}
-	log.Infof("getTestStatusFromBigQuery completed in %s with %d sample results and %d base results from db", time.Since(before), len(sampleStatus), len(baseStatus))
+	report := c.generateComponentTestReport(componentReportTestStatus.BaseStatus, componentReportTestStatus.SampleStatus)
+	report.GeneratedAt = componentReportTestStatus.GeneratedAt
+	return report, nil
+}
 
-	report := c.generateComponentTestReport(baseStatus, sampleStatus)
+func (c *componentReportGenerator) GenerateComponentReportTestStatus() (apitype.ComponentReportTestStatus, []error) {
+	before := time.Now()
+	componentReportTestStatus, errs := c.getTestStatusFromBigQuery()
+	if len(errs) > 0 {
+		return apitype.ComponentReportTestStatus{}, errs
+	}
+	log.Infof("getTestStatusFromBigQuery completed in %s with %d sample results and %d base results from db", time.Since(before), len(componentReportTestStatus.SampleStatus), len(componentReportTestStatus.BaseStatus))
 	now := time.Now()
-	report.GeneratedAt = &now
-	return report, nil
+	componentReportTestStatus.GeneratedAt = &now
+	return componentReportTestStatus, nil
 }
 
-func (c *componentReportGenerator) GenerateTestDetailsReport() (apitype.ComponentReportTestDetails, []error) {
+func (c *componentReportGenerator) GenerateTestDetailsReport(cc cache.Cache, cacheKey interface{}) (apitype.ComponentReportTestDetails, []error) {
```

Comment:
cache should probably just go into `componentReportGenerator` struct as a private field.  And don't we know the type of cacheKey for this function (instead of `interface{}`)?